### PR TITLE
Handlers and Middlewares processing union

### PR DESCRIPTION
**Pull request by Badiboy**
## Description
Call for handlers now union in a single function for future extension.
Plus minor fixes in storages.

## Checklist:
- [x] My changes won't break backward compatibility

### CODE REVIEW

File: telebot/__init__.py [hunk 1]
```diff
@@ -5875,7 +5876,7 @@ def _check_filter(self, message_filter, filter_value, message):
             return False
 
     # middleware check-up method
-    def _check_middleware(self, update_type):
+    def _check_middlewares(self, update_type):
```
Comment (coder2020official) [inline telebot/__init__.py:5879]: I think we can rename this function to _get_middlewares

File: telebot/__init__.py [hunk 2]
```diff
@@ -5889,100 +5890,111 @@ def _check_middleware(self, update_type):
 
     def _run_middlewares_and_handler(self, message, handlers, middlewares, update_type):
```
Comment (coder2020official) [inline telebot/__init__.py:5891]: And this function should be renamed to smth like _run_update_task

**Review by coder2020official — CHANGES_REQUESTED** [review]
~ I will test it tonight.

**Comment by coder2020official** [comment]
- This pull request will:
- help to #1704 (by making it possible to get the result of a handler without middlewares;
- Improve the readability of code

File: telebot/__init__.py (hunk 1 above)
Comment (Badiboy) [reply telebot/__init__.py]: +

File: telebot/__init__.py (hunk 2 above)
Comment (Badiboy) [reply telebot/__init__.py:5891]: Don't like this name. Let's left it for next updates.

**Comment by byehack** [comment]
@Badiboy @coder2020official 
I think by #1706 we can deprecate middleware functions, and remove it in next major version.
You can open a draft PR to start deprecating middleware part.
if you still know a usecase of middlewares after merging #1706 , I would happy to know.

**Comment by coder2020official** [comment]
Middleware functions are another way of using middlewares. We are not going to deprecate it for now

**Pull request merged**